plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2988,6 +2997,12 @@ bool producer_plugin_impl::push_read_only_transaction(transaction_metadata_ptr t
       if( !_ro_in_read_only_mode ) {
          chain.set_db_read_only_mode();
       }
+
+      if ( app().executor().is_write_window() ) {
+         auto idle_time = fc::time_point::now() - _idle_trx_time;
+         _time_tracker.add_idle_time( idle_time );
```

Comment:
It looks like idle time is not tracked for a specific transaction type (read_only or transient), but for both. So it seems to me that this code will count the read_window time as idle. Is is what we want?

Reply:
This idle time is idle time of the main thread that is producing a block (or speculatively executing trxs into a ephemeral block), so yes.